src/common/dcsvg.cpp

```diff
@@ -108,6 +118,151 @@ wxString wxBrushString(wxColour c, int style = wxBRUSHSTYLE_SOLID)
     return s;
 }
 
+wxString wxGetPenPattern(wxPen& pen)
+{
+    wxString s;
+    float w = pen.GetWidth();
+    if (pen.GetWidth() == 0)
+        w = 1;
+    w = w / 3;
+
+    switch (pen.GetStyle())
+    {
+        case wxPENSTYLE_DOT:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f\" "), w * 2, w * 5);
+            break;
+        case wxPENSTYLE_SHORT_DASH:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f\" "), w * 10, w * 8);
+            break;
+        case wxPENSTYLE_LONG_DASH:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f\" "), w * 15, w * 8);
+            break;
+        case wxPENSTYLE_DOT_DASH:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f,%f,%f\" "), w * 8, w * 8, w * 2, w * 8);
+            break;
+        case wxPENSTYLE_USER_DASH:
+        {
+            s = wxT("stroke-dasharray=\"");
+            wxDash *dashes;
+            int count = pen.GetDashes(&dashes);
+            if ((dashes != NULL) && (count > 0))
+            {
+                for (int i = 0; i < count; ++i) {
+                    s << dashes[i];
+                    if (i < count - 1)
+                        s << ",";
+                }
+            }
+            s += wxT("\" ");
+            break;
+        }
+        default:
+            break;
+    }
+    return s;
+}
+
+wxString wxGetBrushFill(wxBrush& brush)
+{
+    wxString s;
+
+    switch (brush.GetStyle())
+    {
+        case wxBRUSHSTYLE_FDIAGONAL_HATCH:
+        {
+            wxString patternName = wxT("FdiagonalHatch") + brush.GetColour().GetAsString(wxC2S_HTML_SYNTAX).substr(1);
```

Comment:
It's a bit painful to repeat all this code, couldn't we just initialize `patternName` in the switch and do everything else outside of it? We seem to be doing the same thing for all cases or am I missing something?